src/Moq/MethodCallReturn.cs

```diff
@@ -171,6 +171,13 @@ private void ValidateReturnDelegate(Delegate callback)
 		{
 			var callbackMethod = callback.GetMethodInfo();
 
+			var expectedParams = this.Method.GetParameters();
+			var actualParams = callback.GetMethodInfo().GetParameters();
```

Comment:
You aren't using `actualParams` anywhere... are you? Looks like a copy & paste error.